router/batchrouter/asyncdestinationmanager/snowpipestreaming/snowpipestreaming.go

```diff
@@ -187,6 +214,12 @@ func (m *Manager) Upload(asyncDest *common.AsyncDestinationStruct) common.AsyncU
 	for _, info := range uploadInfos {
 		imInfo, discardImInfo, err := m.sendEventsToSnowpipe(ctx, asyncDest.Destination.ID, &destConf, info)
 		if err != nil {
+			var authzErr *snowpipeAuthzError
+			if errors.As(err, &authzErr) {
+				if shouldResetBackoff {
+					shouldResetBackoff = false
+				}
+			}
```

Comment:
```suggestion
			if errors.As(err, &authzErr) && !shouldResetBackoff {
				shouldResetBackoff = false
			}
```

Reply:
Removed `shouldResetBackoff` since it was unnecessary